src/opera/test/pge/test_metfile.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2021, by the California Institute of Technology.
+# ALL RIGHTS RESERVED.
+# United States Government sponsorship acknowledged.
+# Any commercial use must be negotiated with the Office of Technology Transfer
+# at the California Institute of Technology.
+# This software may be subject to U.S. export control laws and regulations.
+# By accepting this document, the user agrees to comply with all applicable
+# U.S. export laws and regulations. User has the responsibility to obtain
+# export licenses, or other export authority as may be required, before
+# exporting such information to foreign countries or providing access to
+# foreign persons.
+#
+
+"""
+=================
+test_metfile.py
+=================
+
+Unit tests for the util/metfile.py module.
+"""
+import fileinput
+import os
+import shutil
+import tempfile
+import unittest
+
+from os.path import abspath, join
+
+from pkg_resources import resource_filename
+
+from opera.pge import PgeExecutor, RunConfig
+from opera.util.metfile import MetFile
+from opera.util import PgeLogger
+
+
+class MetFileTestCase(unittest.TestCase):
+    """Base test class using unittest"""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """
+        Set up directories for testing
+        Initialize regular expression
+        Initialize other class variables
+
+        """
+        cls.starting_dir = abspath(os.curdir)
+        cls.test_dir = resource_filename(__name__, "")
+        cls.data_dir = join(cls.test_dir, "data")
+
+        os.chdir(cls.test_dir)
+
+        cls.working_dir = tempfile.TemporaryDirectory(
+            prefix="test_time_", suffix='temp', dir=os.curdir)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        """
+        At completion re-establish starting directory
+        -------
+        """
+        cls.working_dir.cleanup()
+        os.chdir(cls.starting_dir)
+
+    def setUp(self) -> None:
+        """
+        Use the temporary directory as the working directory
+        -------
+        """
+        os.chdir(self.working_dir.name)
+
+    def tearDown(self) -> None:
+        """
+        Return to starting directory
+        -------
+        """
+        os.chdir(self.test_dir)
+
+    def testMetFile(self):
+        """
+        Tests instantiation of MetFile class
+        Methods tested in MetFile class:
+            set_key_value()
+            write_met_file()
+            read_met_file()
+
+        """
+        met_file = './testMetFile.met'
```

Comment:
Use `resource_filename` to get the location of this file.

Reply:
Done